manifest-build-tools/application/reprove.py

```diff
@@ -243,15 +426,38 @@ def main():
 
     if args.git_credential:
         manifest_actions.set_git_credentials(args.git_credential)
-
+    
     if args.jobs:
         manifest_actions.set_jobs(args.jobs)
 
+    if args.branch_name:
+        manifest_actions.set_branch_name(args.branch_name)
+
+    manifest_actions.check_builddir()
     # Start to check out a set of repositories within a manifest file
     if 'checkout' in manifest_actions.actions:
-        manifest_actions.check_builddir()
         manifest_actions.get_repositories()
 
+    if 'branch' in manifest_actions.actions:
+        new_branch = manifest_actions.get_branch_name()
+        if 'checkout' not in manifest_actions.actions:
+            manifest_actions.get_repositories()
+
+        if new_branch is not None:
```

Comment:
when do ```branch_existing_repositories```, 
```
checkout_branch_repositories 
update_package_references 
push_changed_repositories
```
must be done afterwards.
So why don't combine this invoking into ```branch_existing_repositories```?




F.Y.I or moreover, wrap all this funcitons which can run independently into "actions".
create a dict of action contains all dependency relation ships.
Then run each action according to this dict.